### PR TITLE
app-emulation/playonlinux: bump version to 4.3.4; remove netcat6 from RDEPEND

### PR DESCRIPTION
**Pull request by steils**
app-emulation/playonlinux: bump version to 4.3.4

Closes: https://bugs.gentoo.org/672944
Package-Manager: Portage-2.3.62, Repoman-2.3.12
Signed-off-by: Stefan Strogin <stefan.strogin@gmail.com>

app-emulation/playonlinux: remove netcat6 from RDEPEND
    
Remove obsolete net-analyzer/netcat6 from RDEPEND, add
net-analyzer/openbsd-netcat as an alternative to net-analyzer/netcat.

Bug: https://bugs.gentoo.org/573814
Package-Manager: Portage-2.3.62, Repoman-2.3.12
Signed-off-by: Stefan Strogin <stefan.strogin@gmail.com>

### CODE REVIEW

File: app-emulation/playonlinux/playonlinux-4.3.4.ebuild [hunk 1]
```diff
@@ -0,0 +1,117 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python2_7 )
+
+inherit gnome2-utils python-single-r1
```
Comment (juippis) [inline app-emulation/playonlinux/playonlinux-4.3.4.ebuild]: Use xdg or xdg-utils for icon_cache_thing please. Should allow later EAPI to be used, too.

File: app-emulation/playonlinux/playonlinux-4.3.4.ebuild [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+IUSE="winbind"
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+
+DEPEND=""
+RDEPEND="${PYTHON_DEPS}
+	app-arch/cabextract
+	app-arch/p7zip
+	app-arch/unzip
+	app-crypt/gnupg
+	virtual/wine
```
Comment (juippis) [inline app-emulation/playonlinux/playonlinux-4.3.4.ebuild]: Please sort this list alphabetically too while at it.

File: app-emulation/playonlinux/playonlinux-4.3.4.ebuild [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+	doicon -s 128 etc/${PN}.png
+	for size in 16 22 32; do
+		newicon -s $size etc/${PN}$size.png ${PN}.png
+	done
+
+	doman "${FILESDIR}"/playonlinux{,-pkg}.1
+	dodoc CHANGELOG.md
+
+	make_wrapper ${PN} "./${PN}" "/usr/share/${PN}"
+	make_wrapper ${PN}-pkg "./${PN}-pkg" "/usr/share/${PN}"
```
Comment (juippis) [inline app-emulation/playonlinux/playonlinux-4.3.4.ebuild]: Doesn't this come from eutils? Some eclass is probably inheriting it for you, but you should imply it's being used by inheriting it by yourself.

File: app-emulation/playonlinux/playonlinux-4.3.4.ebuild [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+	make_desktop_entry ${PN} ${MY_PN} ${PN} Game
+}
+
+pkg_preinst() {
```
Comment (juippis) [inline app-emulation/playonlinux/playonlinux-4.3.4.ebuild]: This phase can be removed, _savelist is called automatically when xdg_icon_cache_update is being used.

**Comment by juippis** [comment]
And since np-hardass is on devaway / can't commit now, I think you need to find someone to commit this for you. @chewi interested?

File: app-emulation/playonlinux/playonlinux-4.3.4.ebuild (hunk 1 above)
Comment (steils) [reply app-emulation/playonlinux/playonlinux-4.3.4.ebuild]: xdg-utils, agreed.

File: app-emulation/playonlinux/playonlinux-4.3.4.ebuild (hunk 3 above)
Comment (steils) [reply app-emulation/playonlinux/playonlinux-4.3.4.ebuild]: Yes, it was inherited implicitly till EAPI 6.

File: app-emulation/playonlinux/playonlinux-4.3.4.ebuild (hunk 4 above)
Comment (steils) [reply app-emulation/playonlinux/playonlinux-4.3.4.ebuild]: Right. Thanks.

**Comment by steils** [comment]
Thanks for the review @juippis.
I changed the ebuild according to juippis' suggestions, and added app-misc/jq to RDEPEND (since playonlinux-4.3 requires it) and bumped EAPI to 7.

**Comment by Chiitoo** [comment]
At this time, I have not tested the ebuild (never really used the application at all), but it looks fine to me.

What do you think, @sarnex?

**Comment by Chiitoo** [comment]
> At this time, I have not tested the ebuild [...]

Ended up doing a build and run test, which I did not notice any problems with either.

**Comment by sarnex** [comment]
Looks good. @steils can you please squash all of your commits into one, then I'll pull.

Thanks,
Sarnex

**Comment by steils** [comment]
Thanks everybody for the review.

@sarnex 

> can you please squash all of your commits into one, then I'll pull.

Done so.

File: app-emulation/playonlinux/playonlinux-4.3.4.ebuild [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+	app-arch/unzip
+	app-crypt/gnupg
+	app-misc/jq
+	dev-python/wxpython:3.0[${PYTHON_USEDEP}]
+	media-gfx/icoutils
+	|| ( net-analyzer/netcat net-analyzer/openbsd-netcat )
+	net-misc/wget
+	virtual/imagemagick-tools
+	virtual/wine
+	winbind? ( net-fs/samba[winbind] )
```
Comment (juippis) [inline app-emulation/playonlinux/playonlinux-4.3.4.ebuild]: When sorting these dependency lists, remember that unconditional deps come before conditional deps. And in lexicographic ordering the `||` is probably above everything else...

Comment (steils) [reply app-emulation/playonlinux/playonlinux-4.3.4.ebuild]: What makes you say so? I do not see any specific instructions or recommendations on this neither in devmanual nor in skel.ebuild.
The examples from https://devmanual.gentoo.org/quickstart/index.html and https://devmanual.gentoo.org/general-concepts/dependencies/ do not follow your recommendation. I would be grateful for a citation.

In this case I would like to leave it as it is until maintainter's comment. Or a developer that is going to push this commit can reorder the dependencies in a way he likes.

Comment (sarnex) [reply app-emulation/playonlinux/playonlinux-4.3.4.ebuild]: I'm not totally sure what juippis is talking about to be honest, I personally prefer it in lexigraphical order which it seems to be.

Comment (juippis) [reply app-emulation/playonlinux/playonlinux-4.3.4.ebuild]: Well it's something that's always asked to "fix" by proxy-maint devs, 
https://github.com/gentoo/gentoo/pull/9045#discussion_r200806099
so I guess it just got stuck with me. Could be styling prefence in the end. 

Seems like I was wrong about the position of `||` in lexicographic ordering. It doesn't come first, it comes last. 
```
$ cat tmp1.txt 
${PYTHON_DEPS}
app-arch/cabextract
app-arch/p7zip
app-arch/unzip
app-crypt/gnupg
app-misc/jq
dev-python/wxpython:3.0[${PYTHON_USEDEP}]
media-gfx/icoutils
|| ( net-analyzer/netcat net-analyzer/openbsd-netcat )
net-misc/wget
virtual/imagemagick-tools
virtual/wine
winbind? ( net-fs/samba[winbind] )
x11-apps/mesa-progs
x11-terms/xterm

$ LC_ALL=C sort tmp1.txt 
${PYTHON_DEPS}
app-arch/cabextract
app-arch/p7zip
app-arch/unzip
app-crypt/gnupg
app-misc/jq
dev-python/wxpython:3.0[${PYTHON_USEDEP}]
media-gfx/icoutils
net-misc/wget
virtual/imagemagick-tools
virtual/wine
winbind? ( net-fs/samba[winbind] )
x11-apps/mesa-progs
x11-terms/xterm
|| ( net-analyzer/netcat net-analyzer/openbsd-netcat )
```

**Pull request closed without merging**